Improve provision and key mgmt implementation

rajkan01 opened the pull request:
- Add key set status API to set the key context status.
- Extend the provision store implementation to support COSE certificate storage to PS.
- csr callback request decodes the csr function and updates the provision data structure accordingly, it's defined as the local context variable, so update the provision data structure prov_present flag accordingly.
- Extend the provision store function and also updtate the key status context along with storing the received certificate to PS.
- The secure and Non-secure sides maintain the status of the key context, and these changes are to extend the existing huk key ec stat service can set the key status from the non-secure side for setting X509 key cert status once the certificate is generated on non-secure side.
- Strip down the x509 certificate gen to the short form to print key status command print in aligned table format.
-

Commit:
sample:key_mgmt: Add new key context set status API

Add key set status API to set the key context status.

Signed-off-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@linaro.org>

Commit:
sample: provision: Add COSE certificate store to PS support

Extend the provision store implementation to support COSE certificate
storage to PS.

Signed-off-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@linaro.org>

Commit:
sample: bootstrap: Fix provision present flag handling

csr callback request decodes the csr function and updates the provision
data structure accordingly, it's defined as the local context variable,
so update the provision data structure prov_present flag accordingly.

Signed-off-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@linaro.org>

Commit:
sample: bootstrap: update provision store to update key status

Extend the provision store function and also updtate the key status
context along with storing the received certificate to PS.

Signed-off-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@linaro.org>

Commit:
sample: huk: update huk EC key status service

The secure and Non-secure sides maintain the status of the key context,
and these changes are to extend the existing huk key ec stat service can
set the key status from the non-secure side for setting X509 key cert
status once the certificate is generated on non-secure side.

Signed-off-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@linaro.org>

Commit:
sample: shell: strip down the x509 certificate gen to the short form.

Signed-off-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@linaro.org>

microbuilder reviewed (changes requested):
@rajkan01 Thanks for the PR!

Looks good, but two things occurred to me testing this out:

1. We can display the public key via `keys public <id>` but do you think it's useful to also add a `keys cert <id>` command to display the certificate data for testing/debugging/verification?

2. Small typo in the help menu here (`$ keys ca ca <Key ID>`) that we can probably fix in this PR:

```
uart:~$ keys ca
argc: 1
Request certificate from bootstrap server for the given key

$ keys ca ca <Key ID>

Run 'status' for key ID list

Example: $ keys ca 5001
```

Commit:
sample: Correct the file format

Signed-off-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@linaro.org>

Commit:
sample: shell: Add keys cert command support

Add the keys cert <key id> command that helps the user retrieve the
certificate and print it on the console.

Signed-off-by: Rajkumar Kanagaraj <rajkumar.kanagaraj@linaro.org>
